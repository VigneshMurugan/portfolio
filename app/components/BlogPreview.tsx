import classnames from "classnames";
import { Link } from "remix";

export interface BlogPreviewProps {
  className?: string;
  content: string;
  slug: string;
  title: string;
  image: string;
}

export const BlogPreview = (props: BlogPreviewProps) => {
  const { className, content, slug, image, title } = props;

  // Styles
  const tailwind = `flex flex-col aspect-video---`;
  const cssComponent = classnames("blog-preview", tailwind, className);

  return (
    <Link className={cssComponent} to={`/blog/${slug}`}>
      <h3 className="my-4 text-2xl">{title}</h3>
      {/* <img alt="" loading="lazy" src={image} /> */}
      {/* <div dangerouslySetInnerHTML={{ __html: content }} /> */}
    </Link>
  );
};
