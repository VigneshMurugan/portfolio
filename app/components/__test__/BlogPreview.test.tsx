import { expect, test } from "vitest";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import { BlogPreview } from "../BlogPreview";

describe("BlogPreview Component", () => {
  const component = render(
    <MemoryRouter>
      <BlogPreview id={10} image="" title="test title" />
    </MemoryRouter>
  );

  test("should render our BlogPreview", () => {
    expect(component.baseElement).toMatchSnapshot();
  });
});
