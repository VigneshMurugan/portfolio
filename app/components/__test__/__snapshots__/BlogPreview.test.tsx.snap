// Vitest Snapshot v1

exports[`BlogPreview Component > should render our BlogPreview 1`] = `
<body>
  <div>
    <a
      class="blog-preview flex flex-col aspect-video---"
      href="/blog/undefined"
    >
      <img
        alt=""
        class="border-color-border-- border- mt-4"
        loading="lazy"
        src=""
      />
      <h3
        class="text-highlight mt-4 text-2xl"
      >
        test title
      </h3>
      <div>
        Invalid Date
      </div>
    </a>
  </div>
</body>
`;
